### ISSUE TITLE
Overall slowness

### ISSUE THREAD

**Issue by 3esmit** [files-community/Files#16191]
### Description

I really liked the software, it's an amazing improvement over built-in Windows Explorer.

I wanted to use it as replacement for the Windows Explorer, however the slowness of every operation would prevent me. 

### Steps To Reproduce

Deleting a files:
Windows explorer is instant, press delete, it vanishes the file. Files, it have a huge delay (5 seconds), even for starting to delete, and then for deleting.
If deleting many files, Windows explorer delete all of them, and than refresh, Files refresh after every single deleted file, and every refresh is very slow. 

Navigating through folders is painful:
In Windows explorer, click a folder, instantly opens, go back, instantly goes back.
In Files it have loading time for opening and going back. The more files/folders in the directory, the slower it is. 

### Requirements

Operations must be instant, as in Windows Explorer. 

### Files Version

3.7.6.0

### Windows Version

Windows 10 Pro 22H2 19045.4780 Windows Feature Experience Pack 1000.19060.1000.0

### Log File

[debug.log](https://github.com/user-attachments/files/16998181/debug.log)

**Comment by 0x5bfa** [comment]
Thank you for your feedback.

> Deleting a files

We're all aware of the issues. Primary obstacle regarding the issue we use very very complicated codebase around the file operations. We've been trying to simply the codebase including status reporting and P/Invoke'ing native APIs.

> Navigating through folders is painful

We've taken slowness during page navigation very seriously and fixed it to some extent while we think there's still room to improvement.

Basically we track the code optimizations and performance improvements in #4180

**Comment by yaira2** [comment]
> Windows explorer is instant, press delete, it vanishes the file. Files, it have a huge delay (5 seconds), even for starting to delete, and then for deleting.

Thank you so much for your feedback! We’re aware of an issue where deleting items can be a bit slow when the Recycle Bin already has a lot of items. While we work on a proper fix, you can workaround this by emptying the Recycle Bin. Thanks for your patience! #14108 

> In Files it have loading time for opening and going back. The more files/folders in the directory, the slower it is.

Loading items in Files should be quite similar to File Explorer, though they might appear a bit slower because of the newer UI framework. Could you please provide more details about the slowness you’re experiencing?

**Comment by 3esmit** [comment]
I think the best would be a video showing the behavior, so I recorded the task I was doing that felt painful. 
[2024-09-15 20-41-39.mkv.zip](https://github.com/user-attachments/files/17007779/2024-09-15.20-41-39.mkv.zip)

For other folders is not that slow, but for this task in particular (cleaning up the uninstalled program's zombie shortcuts from start menu) I had anxiety.

**Comment by yaira2** [comment]
> Notice that not only its slow to load, it loads first some part, and than the rest, and doing so it changes the position of the first loaded stuff, making this slowness even worst. I would prefer that it be more slower but show the contents all at once, or show them as they are processed, but dont change the position of what was loaded first.

Startup performance is being worked on but in the meantime, you can try the option to keep Files open in the background.

**Comment by pooxid** [comment]
I was about to write a report but instead I'll drop +1 here. I love the soft but is quite slugish on my system especially navigating through folders. Relatively half the speed of windows explorer if not slower.

**Comment by 3esmit** [comment]
> > Notice that not only its slow to load, it loads first some part, and than the rest, and doing so it changes the position of the first loaded stuff, making this slowness even worst. I would prefer that it be more slower but show the contents all at once, or show them as they are processed, but dont change the position of what was loaded first.
> 
> Startup performance is being worked on, but in the meantime, you can try the option to keep Files open in the background.

I use that feature, its not the startup slowness, but the navigation when there are many folders in the directory I am going exploring.

The case is

I have a directory, this directory have like 50 folders inside, and I want to go inside each of them to see whats inside, and go back to the parent folder. This is where it gets too slow to work smoothly.

**Comment by yaira2** [comment]
@3esmit thank you for the clarification, we'll continue working on improvements to performance.

**Comment by pooxid** [comment]
> > I was about to write a report but instead I'll drop +1 here. I love the soft but is quite slugish on my system especially navigating through folders. Relatively half the speed of windows explorer if not slower.
> 
> Can you clarify which part is slow? For example, loading the file names, loading the icons etc.

Sure, there are multiple stages where delays between action and execution may appear. 
1. After double clicking to open a folder there is a semi linear waiting time before entering a folder in relation to how many files/folder are inside. 
2. Sometimes after entering a folder there may be another loading time with a flat empty windows (approximately 1 sec) until file/folder names appear. On their visual arrival the listing of the names is fast.
3. Loading of the icons is slow across the board.

At some point there was additional waiting time from a listing issue like @3esmit was experiencing coming from Settings/Layout/Sorting and grouping being different from the active windows Sorting and grouping outside of the global settings. This led to double rearrangement/listing executed one after another. If they are synced there is no issue.

My system is quiet dirty at the moment and it may be related to my sluggish experience, but for reference Windows explorer and Dolphin file manager are working as expected (Dolphin being exceptionally fast)
